src/coreclr/src/jit/morph.cpp

```diff
@@ -10426,15 +10426,15 @@ GenTree* Compiler::fgMorphCopyBlock(GenTree* tree)
                             }
                         }
                     }
-                    else // if (lvaGetDesc(fieldLclNum)->lvFldOffset != 0)
-                    {
-                        srcFld = gtNewOperNode(GT_ADD, TYP_BYREF, srcFld,
-                                               new (this, GT_CNS_INT)
-                                                   GenTreeIntCon(TYP_I_IMPL, lvaGetDesc(dstFieldLclNum)->lvFldOffset,
-                                                                 curFieldSeq));
-                    }
+
                     if (!done)
                     {
+                        // Add the fieldSeq offset for all fields, even for 0 offset.
+                        // We will read these field seq when we ask for `CORINFO_CLASS_HANDLE` for these trees.
+                        // We can use `fieldSeq` from the dst because fieldHnd for src and dst must match.
+                        GenTreeIntCon* fldOffset = new (this, GT_CNS_INT)
```

Comment:
Looks like you could shorten this a bit by using the `gtNewIconNode(unsigned fieldOffset, FieldSeqNode* fieldSeq)` overload I added a while ago.

Reply:
That helped me to find another case where we did not create if the offset was zero:
https://github.com/dotnet/runtime/blob/71d2911c64058434a5db6b324c7efead93dfde4c/src/coreclr/src/jit/morph.cpp#L6295

the strange thing is that it was created with `TYP_INT` instead of `TYP_I_IMPL`. probably does't matter, but I will check the diffs.